src/tests/system/tests/test_kcm.py

```diff
@@ -413,3 +415,335 @@ def test_kcm__simple_kinit(client: Client, kdc: KDC):
+        with client.auth.kerberos(ssh) as krb:
+            krb.kdestroy()
+
+    end_log_nodebug = kcm_log_length()
+    assert start_log_length == end_log_nodebug, "Debug messages were generated"
+
+    client.sssd.kcm["debug_level"] = "9"
+    client.sssd.config_apply()
+    assert client.svc.restart("sssd-kcm").rc == 0, "Restart of kcm failed"
+
+    with client.ssh(user, password) as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            krb.kdestroy()
+
+    end_log_debug = kcm_log_length()
+    assert start_log_length + 100 < end_log_debug, "Debug messages were not generated"
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_kcm__ssh_login_with_kcm(client: Client, ldap: LDAP, kdc: KDC):
+    """
+    :title: kcm: Verify ssh login is successuful with kcm as default
+    :setup:
+        1. Add user and principal
+        2. Set kerberos as default auth provider
+        3. Start SSSD
+    :steps:
+        1. Authenticate as "user1" over SSH using kcm
+    :expectedresults:
+        1. Authenticated successfully
+    :customerscenario: False
+    """
+    ldap.user("user1").add()
+    kdc.principal("user1").add()
+
+    client.sssd.common.krb5_auth(kdc)
+    client.sssd.start()
+
+    with client.ssh("user1", "Secret123") as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            res = krb.klist()
+            assert res.rc == 0, "Klist failed"
+
+
+@pytest.mark.ticket(gh=4763)
+@pytest.mark.topology(KnownTopology.Client)
+def test_kcm__display_correct_kvno(client: Client, kdc: KDC):
+    """
+    :title: kvno correctly displays version numbers of principals
+    :setup:
+        1. Add Kerberos principal "user1" to KDC
+        1. Add Kerberos service principal "host/client" to KDC
+        2. Add local user "user1"
+        3. Start SSSD
+    :steps:
+        1. Authenticate as "user1" over SSH
+        2. Kinit as "user1"
+        3. Call kvno on service principal
+        4. Check output of kvno
+    :expectedresults:
+        1. Authenticated successfully
+        2. Kinit was successful
+        3. Kvno was successful
+        4. Output is correct
+    :customerscenario: True
+    """
+    user = "user1"
+    password = "Secret123"
+
+    kdc.principal(user).add(password=password)
+    kdc.principal("host/client").add(password=None)
+    client.local.user(user).add(password=password, uid=99000)
+    client.sssd.common.kcm(kdc)
+    client.sssd.start()
+
+    with client.ssh(user, password) as ssh:
+        with client.auth.kerberos(ssh) as krb:
+            krb.kinit(user, password=password)
+            res = krb.kvno("host/client", realm="TEST")
+            assert res.rc == 0
+
+            for line in res.stdout_lines:
+                kvno_check = re.search(r"host/client@TEST: kvno = (\d+)", line)
+                assert kvno_check, f"kvno display was not correct, line: {line}"
+
+
+@pytest.mark.topology(KnownTopology.Client)
+def test_kcm__per_uid_quota(client: Client, kdc: KDC):
```

Comment:
Let's merge the last remaining test cases and call it

test_kcm__configure_max_uid_ccaches_with_different_values()

1. test the default 64
2. test a lower value 1
3. test a higher value 65

It's not so important with the KCM provider, but I'd like this to be the new test standard, instead of a lot of little tests, a single, more comprehensive test.